scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -763,5 +766,386 @@
+        offer-4 (offer-maker 4 2048 0)
+        offer-5 (offer-maker 4 1024 0)
+        offer-6 (offer-maker 10 4096 10)
+        offer-7 (offer-maker 20 4096 5)
+        offer-8 (offer-maker 30 16384 1)
+        run-handle-resource-offers! (fn [num-considerable offers & {:keys [user-quota user->usage]}]
+                                      (reset! launched-offer-ids-atom [])
+                                      (reset! launched-task-ids-atom [])
+                                      (let [conn (restore-fresh-database! uri)
+                                            test-db (d/db conn)
+                                            driver-atom (atom nil)
+                                            ^TaskScheduler fenzo (sched/make-fenzo-scheduler driver-atom 1500 0.8)
+                                            group-ent-id (create-dummy-group conn)
+                                            job-1 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 3 :memory 2048))
+                                            job-2 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 13 :memory 1024))
+                                            job-3 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 7 :memory 4096))
+                                            job-4 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 11 :memory 1024))
+                                            job-5 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 5 :memory 2048 :gpus 2))
+                                            job-6 (d/entity test-db (create-dummy-job conn :group group-ent-id :ncpus 19 :memory 1024 :gpus 4))
+                                            category->pending-jobs-atom (atom {:normal [job-1 job-2 job-3 job-4]
+                                                                               :gpu [job-5 job-6]})
+                                            user->usage (or user->usage {test-user {:count 1, :cpus 2, :mem 1024, :gpus 0}})
+                                            user->quota (or user-quota {test-user {:count 10, :cpus 50, :mem 32768, :gpus 10}})
+                                            result (sched/handle-resource-offers! conn driver fenzo fid category->pending-jobs-atom user->usage user->quota
+                                                                                  num-considerable offers-chan offers)]
+                                        (async/>!! offers-chan :end-marker)
+                                        result))]
+
+    (testing "enough offers for all normal jobs"
+      (let [num-considerable 10
+            offers [offer-1 offer-2 offer-3]]
+        (is (run-handle-resource-offers! num-considerable offers))
+        (is (= :end-marker (async/<!! offers-chan)))
+        (is (= 3 (count @launched-offer-ids-atom)))
+        (is (= (+ 3 13 7 11) (reduce + (map :cpus @launched-task-ids-atom))))
```

Comment:
You can set a job-name with create-dummy job and then get the name from a task (`(get-in task [:job :job/name])`). Then you can use the job name to identify jobs in your assertions, instead of using prime amounts of CPUs as identifiers.